helix-core/src/main/java/org/apache/helix/controller/stages/TaskGarbageCollectionStage.java

```diff
@@ -73,9 +73,12 @@ public void process(ClusterEvent event) throws Exception {
           continue;
         }
         long purgeInterval = workflowConfig.getJobPurgeInterval();
+        if (purgeInterval <= 0) {
+          continue;
+        }
         long currentTime = System.currentTimeMillis();
         long nextPurgeTime = workflowContext.getLastJobPurgeTime() + purgeInterval;
-        if (purgeInterval > 0 && nextPurgeTime <= currentTime) {
```

Comment:
why the previous way has a bug?

Reply:
Since we allow negative delay in scheduleRebalance, which leads to immediate rebalance (on-demand), the current logic is incorrect. When job purge interval is non-positive, a rebalance is always triggered immediately.

See after this if statement for the rebalance scheduling line. That line is always triggered in the previous version.

Reply:
I see due to this line:
```
        scheduleNextJobPurge(workflowConfig.getWorkflowId(), nextPurgeTime, _rebalanceScheduler,
            manager);
```

previously it always schedule the purge. 

Help me to understand the impact of this a little bit more. With this change, event this is not always scheduled. From the perspective of flaky test, it does not matter, right? I mean with this pull change, we will still see the flaky test to be flaky.